src/index.md

```diff
@@ -1445,6 +1445,48 @@ Disadvantages of this model are that the list of backports in JBS will still lis
+
+## Rebase before creating the PR
+
+It's likely that other people have pushed changes to the code base since you created your branch. Make sure to pull the latest changes and rebase your fix on top of that before creating your PR. This is a courtesy issue. Your reviewers shouldn't have to read your patch on top of old code that has since changed. This is hopefully obvious in cases where the upstream code has gone through cleanups or refactorings, and your patch may need similar cleanups in order to even compile. But even in cases where only smaller changes has been done, the reviewers shouldn't have to react to issues like "that line of code was moved last week, why is it back there?".
+
+~~~
+git rebase master
+~~~
+
+After the PR has been published, rebasing, force-pushing, and similar actions is strongly discouraged. Such actions will disrupt the workflow for reviewers who fetch the PR branch. Pushing new changes is fine (and even merging if necessary) for a PR under review. Incremental diffs and other tools will help your reviewers see what you have changed. In the end, all commits will be squashed into a single commit automatically, so there're actually no drawbacks what so ever to making several commits to a PR branch during review.
+
+## Life of a PR
+
+#. **Write a useful description**
+
+   The description of the PR should state what problem is being solved and shortly describe how it's solved. Reviewers and other interested readers are referred to the text in the JBS issue for details, but the description should be enough to give an overview.
+
+#. **Make sure all relevant groups are included**
+
+   The bots will make an attempt to include the groups that need to review your change based on the location of the source code you have changed. There may be aspects of your change that are relevant to other groups as well, and the mapping from source to groups isn't always perfect, so make sure all relevant groups have been included, and add new labels using [`/label`](https://wiki.openjdk.java.net/display/SKARA/Pull+Request+Commands#PullRequestCommands-/label) if needed.
+
+#. **Allow enough time for review**
+
+   In general all PRs should be open for at least 24 hours to allow for reviewers in all time zones to get a chance to see it. In some areas [trivial](#trivial) changes are allowed to be pushed without the 24 hour delay. Ask your reviewers if you think this applies to your change.
+
+#. **Merge the latest changes**
+
+   Before pushing you should always fetch and merge the latest changes from the target repository. If your PR is out for review for a longer time it is a good habit to pull from the target repository regularly to keep the change up to date.
```

Comment:
Yeah I don't agree with what was written. There's always changes in the upstream repo. Unless it relates to what I am changing I don't see the need. There can be 100 changes in hotspot and I don't care because it doesn't affect my change to Linux X11 code ..

Reply:
> Unless it relates to what I am changing I don't see the need. There can be 100 changes in hotspot and I don't care because it doesn't affect my change to Linux X11 code ..

Maybe it should be a suggestion?

If one is an experienced contributor, which is unlikely to be true for all the contributors reading the guide, one has good judgement on what is acceptable and what isn't. Because in general, until one runs tests for A+B, one cannot be sure that those tests will pass if they pass for A and B individually.

Reply:
Re-phrased.